File: app/src/main/java/com/leaguechampions/utils/PrefUtils.java
package com.leaguechampions.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.leaguechampions.core.Const;

public class PrefUtils {

    private PrefUtils() {}

    public static boolean isMockMode(Context context) {
        return isMockMode(PreferenceManager.getDefaultSharedPreferences(context));
    }

    public static boolean isMockMode(SharedPreferences sharedPreferences) {
        return sharedPreferences
                .getBoolean(Const.PREF_KEY_MOCK_MODE, false);
    }

    public static void setMockMode(Context context, boolean mockMode) {
        setMockMode(PreferenceManager.getDefaultSharedPreferences(context), mockMode);
    }

    public static void setMockMode(SharedPreferences sharedPreferences, boolean mockMode) {
        sharedPreferences
                .edit()
                .putBoolean(Const.PREF_KEY_MOCK_MODE, mockMode)
                .apply();
    }
}

File: app/src/main/java/com/leaguechampions/utils/PrefUtils.kt
package com.leaguechampions.utils

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

import com.leaguechampions.data.local.Const

object PrefUtils {

    fun isMockMode(context: Context): Boolean {
        return isMockMode(PreferenceManager.getDefaultSharedPreferences(context))
    }

    fun isMockMode(sharedPreferences: SharedPreferences): Boolean {
        return sharedPreferences.getBoolean(Const.PREF_KEY_MOCK_MODE,false)
    }

    fun setMockMode(context: Context, mockMode: Boolean) {
        setMockMode(PreferenceManager.getDefaultSharedPreferences(context), mockMode)
    }

    fun setMockMode(sharedPreferences: SharedPreferences, mockMode: Boolean) {
        sharedPreferences
                .edit()
                .putBoolean(Const.PREF_KEY_MOCK_MODE, mockMode)
                .apply()
    }
}
